auth0/resource_auth0_action.go

```diff
@@ -251,6 +259,61 @@ func deployAction(ctx context.Context, d *schema.ResourceData, m interface{}) di
 	return diag.FromErr(d.Set("version_id", actionVersion.GetID()))
 }
 
+func preventErasingUnmanagedSecrets(d *schema.ResourceData, api *management.Management) diag.Diagnostics {
+	if !d.HasChange("secrets") {
+		return nil
+	}
+
+	preUpdateAction, err := api.Action.Read(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	oldSecrets, newSecrets := d.GetChange("secrets")
+
+	return checkForUnmanagedActionSecrets(
+		oldSecrets.([]interface{}),
+		newSecrets.([]interface{}),
+		preUpdateAction.Secrets,
+	)
+}
+
+func checkForUnmanagedActionSecrets(
+	oldSecretsFromConfig []interface{},
+	newSecretsFromConfig []interface{},
+	secretsFromAPI []*management.ActionSecret,
+) diag.Diagnostics {
+	secretKeysInConfigMap := make(map[string]bool, len(secretsFromAPI))
```

Comment:
I'm still figuring out the logic below, but not sure about setting the length to the number of secrets on remote. Couldn't the number of secrets in old and new be greater than the number on remote, thus making this code susceptible to out-of-bounds errors?

Reply:
If we specify a capacity when using make, that doesn't mean we can't exceed the capacity on items such as slices or maps, so we won't get any out of bounds error. Go behind the scenes knows to automatically increase the size if we need to add more items.